src/charm.py

```diff
@@ -98,68 +99,80 @@ def _configure_pod(self):
-            'containers': [{
-                'name': self.app.name,
-                'imageDetails': image_info,
-                'ports': [{
-                    'containerPort': config['port'],
-                    'protocol': 'TCP'
-                }],
-                'envConfig': {
-                    'MYSQL_ROOT_PASSWORD': config['root-password'],
-                },
-            }]
+            "version": 3,
+            "containers": [
+                {
+                    "name": self.app.name,
+                    "imageDetails": image_info,
+                    "ports": [
+                        {"containerPort": config["port"], "protocol": "TCP"}
+                    ],
+                    "envConfig": self.env_config,
+                }
+            ],
         }
 
         return pod_spec
 
-    def _mysql_is_ready(self):
+    def _mysql_is_ready(self) -> bool:
         """Check that every unit has mysql running.
 
         Until we have a good event-driven way of using the Kubernetes
         readiness probe, we will attempt
         """
-        for hostname in self.hostnames:
-            try:
-                cnx = self._get_sql_connection_for_host(hostname)
-                logger.warning("MySQL service is ready for {}.".format(hostname))
-            except mysql.connector.Error:
-                # TODO: Improve exceptions handling
-                logger.warning("MySQL service is not ready for {}.".format(hostname))
-                return False
-            else:
-                cnx.close()
+        try:
+            cnx = self._get_sql_connection_for_host()
+            logger.info("MySQL service is ready in %s.", self.hostname)
+        except mysql.connector.Error:
+            # TODO: Improve exceptions handling
+            logger.info("MySQL service is not ready in %s", self.hostname)
+            return False
+        else:
+            cnx.close()
 
         return True
 
-    def _get_sql_connection_for_host(self, hostname):
+    def _get_sql_connection_for_host(self):
+        """Helper for the _mysql_is_ready() method"""
         config = {
-            'user': 'root',
-            'password': self.model.config['root-password'],
-            'host': hostname,
-            'port': self.model.config['port']
+            "user": "root",
+            "password": self._stored.MYSQL_ROOT_PASSWORD,
+            "host": self.hostname,
+            "port": self.model.config["port"],
         }
         return mysql.connector.connect(**config)
 
-    def _get_unit_hostname(self, _id: int) -> str:
+    def _get_hostname_from_unit(self, _id: int) -> str:
         """Construct a DNS name for a MySQL unit."""
         return "{0}-{1}.{0}-endpoints".format(self.model.app.name, _id)
 
+    def _get_unit_number_from_unit_name(self, unit_name: str) -> int:
+        """Returns the unit number based in the unit name."""
+        UNIT_RE = re.compile(".+(?P<unit>[0-9]+)")
+        match = UNIT_RE.match(unit_name)
```

Comment:
This doesn't need to be changed as it works. It, however, is probably more efficient (and readable) to use unit_name.split('/')[1]

Reply:
As we use bind_address, the method it's not longer neccesary